File: examples/k2_rnnt2_cli.py
import PyFastASR
import soundfile as sf
import sys
import time

param_path = sys.argv[1]
audio_path = sys.argv[2]


data, samplerate = sf.read(audio_path)
audio_len = data.size / samplerate

print("Audio time is {}s. len is {}.".format(audio_len, data.size))

start_time = time.time()
p = PyFastASR.Model(param_path, 2)
end_time = time.time()

print("Model initialization takes {:.2}s.".format(end_time - start_time))

start_time = time.time()
p.reset()
result = p.forward(data)
end_time = time.time()
print('Result: "{}".'.format(result))
print("Model inference takes {:.2}s.".format(end_time - start_time))

File: examples/k2_rnnt2_cli.cpp
#include <iostream>
#ifndef _WIN32
#include <sys/time.h>
#else
#include <win_func.h>
#endif

#include <Audio.h>
#include <Model.h>

using namespace std;

int main(int argc, char *argv[])
{
    struct timeval start, end;
    Audio audio(1);
    audio.loadwav(argv[2]);
    // audio.padding();
    audio.disp();
    gettimeofday(&start, NULL);
    Model *mm = create_model(argv[1], 2);
    gettimeofday(&end, NULL);
    long seconds = (end.tv_sec - start.tv_sec);
    long micros = ((seconds * 1000000) + end.tv_usec) - (start.tv_usec);
    printf("Model initialization takes %lfs.\n", (double)micros / 1000000);
    audio.split();

    setbuf(stdout, NULL);
    cout << "Result: \"";
    gettimeofday(&start, NULL);
    float *buff;
    int len;
    int flag;
    while (audio.fetch(buff, len, flag) > 0) {
        mm->reset();
        string msg = mm->forward(buff, len, flag);
        cout << msg;
    }

    gettimeofday(&end, NULL);

    cout << "\"." << endl;

    seconds = (end.tv_sec - start.tv_sec);
    micros = ((seconds * 1000000) + end.tv_usec) - (start.tv_usec);
    printf("Model inference takes %lfs.\n", (double)micros / 1000000);

    return 0;
}
